core/src/main/java/org/eclipse/hono/util/CommandRouterResult.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.util;
```

Comment:
is this class required to be in the core module? IMHO it will only ever be needed in the context of the client itself, or am I mistaken?

Reply:
If sticking to the way the other API services are implemented, the Result class is also used in the service interface in the form of `Future<CommandRouterResult>` method return values. That means it's also used in the command router server implementation.

Reply:
Actually, in the client the `CommandRouterResult` isn't really needed. It's only used internally in the `ProtonBasedTenantCommandRouterClient` class and can be replaced with `RequestResponseResult<JsonObject>` there.
So, I've done the replacement and removed the class for now.

Reply:
I did not want to suggest to remove the class altogether but to simply move it to the adapter-amqp module or even make it a private class of ProtonBasedTenantCommandRouterClient. Your choice ...